scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableSyntaxConfig.scala

```diff
@@ -0,0 +1,149 @@
+package scalafix.internal.config
+
+import metaconfig.{Conf, ConfError, ConfDecoder, Configured}
+import org.langmeta._
+import MetaconfigPendingUpstream.XtensionConfScalafix
+
+import scala.meta.tokens.Token
+
+case class DisableSyntaxConfig(
+    carriageReturn: Boolean = false,
+    keywords: Set[DisabledKeyword] = Set(),
+    semicolons: Boolean = false,
+    tabs: Boolean = false,
+    xml: Boolean = false
+) {
+  implicit val reader: ConfDecoder[DisableSyntaxConfig] =
+    ConfDecoder.instanceF[DisableSyntaxConfig](
+      c =>
+        (
+          c.getField(carriageReturn) |@|
+            c.getField(keywords) |@|
+            c.getField(semicolons) |@|
+            c.getField(tabs) |@|
+            c.getField(xml)
+        ).map {
+          case ((((a, b), c), d), e) =>
+            DisableSyntaxConfig(a, b, c, d, e)
+      })
+
+  def isDisabled(keyword: String): Boolean =
+    keywords.contains(DisabledKeyword(keyword))
+}
+
+object DisableSyntaxConfig {
+  val default = DisableSyntaxConfig()
+  implicit val reader: ConfDecoder[DisableSyntaxConfig] = default.reader
+}
+
+object Keyword {
+  def unapply(token: Token): Option[String] = {
+    val prefix = token.productPrefix
+    if (prefix.take(2) == "Kw") {
```

Comment:
We can avoid this string manipulation on every token by precomputing `val keywordProductPrefixes: Set[String] = keywords.toIterator.map(kw => s"Kw${kw.capitalize}").toSet`

Reply:
PS, you can use `String.startsWith`